tests/callbacks/test_eval_output_logging_callback.py

```diff
@@ -193,23 +205,27 @@ def test_eval_output_logging_lm(
     assert f'lm_acc_step_0' in in_memory_logger.tables
     # Only want one table - we log once to a single step value during eval_end()
     assert len(in_memory_logger.tables) == 1
-    assert json.loads(in_memory_logger.tables[f'lm_acc_step_0'])['columns'] == [
+    logged_data = json.loads(in_memory_logger.tables[f'lm_acc_step_0'])
+    assert logged_data['columns'] == [
         'context',
         'label',
         'output',
         'result',
         'metric_name',
+        'outputs',
         'input',
         'run_name',
     ]
+
     # We use the same data in each batch
-    assert json.loads(in_memory_logger.tables[f'lm_acc_step_0'])['data'] == [
+    assert logged_data['data'] == [
         [
             'The dog is',
             ' furry',
             ' furry',
             1,
             'InContextLearningLMAccuracy',
+            RegexMatcher(r' dog is furry(\[PAD\])+I'),
```

Comment:
So I'm not sure you'll like this approach :sweat_smile: 

I actually was surprised to find `output` already in the logged data, since it's not in `eval_output_logger`. Is that from the ICL metric/eval itself? It makes the new `outputs` field look superfluous (though also possibly not, since `outputs` is like a `raw_output`). This pattern matching here to pass the test also feels a bit unprincipled.

Reply:
as for the outputs already being there, thats cause of https://github.com/mosaicml/llm-foundry/blob/21c9e0a58a68b2ddb7cc7bea40a11c9add85d2c9/llmfoundry/callbacks/eval_output_logging_callback.py#L53 right?

Reply:
Yes, I expected `outputs` (with an "s"), but not `output` (not with an "s"). Don't know where that's coming from.

Reply:
oh i see what you mean. Yeah, the logging dict starts off as a deep copy of whatever the metric outputs were (https://github.com/mosaicml/llm-foundry/blob/88511f780c7437e89cdaf96a1771aac41c2a64a5/llmfoundry/callbacks/eval_output_logging_callback.py#L45-L48)